### PR TITLE
Adding margin support to placement.js for <amp-auto-ads>

### PR DESCRIPTION
**Pull request by tlong2**
Margins are specified in the placement object within the JSON config using:

```js
style: {
  top_m: 4,
  bot_m: 5
}
```

### CODE REVIEW

File: extensions/amp-auto-ads/0.1/placement.js [hunk 1]
```diff
@@ -189,7 +197,18 @@ function getPlacementFromObject(win, placementObj) {
...
-  return new Placement(win, anchorElement, injector);
+  let margins = undefined;
+  if (placementObj['style']) {
+    const marginTop = parseInt(placementObj['style']['top_m'], 10);
+    const marginBottom = parseInt(placementObj['style']['bot_m'], 10);
+    if (marginTop) {
+      margins = {top: marginTop};
+    }
+    if (marginBottom) {
+      (margins = margins || {}).bottom = marginBottom;
```
Comment (jridgewell) [inline extensions/amp-auto-ads/0.1/placement.js]: To avoid this, can we just default `margins` to `{ top: 0, bottom: 0 }`?

Comment (tlong2) [inline extensions/amp-auto-ads/0.1/placement.js]: If we set the margins to anything (including 0), then it means the attemptSizeChange logic has to read the existing margins using getComputedStyle (which is expensive), to figure out what the diff is. So if we're not setting any margins it's best to just provide undefined I think.

Comment (jridgewell) [inline extensions/amp-auto-ads/0.1/placement.js]: Or:

```js
if (marginTop || marginBottom) {
  margins = {
    top: marginTop,
    bottom: marginBottom
  };
}

File: extensions/amp-auto-ads/0.1/placement.js [hunk 2]
```diff
@@ -82,8 +82,9 @@ export class Placement {
    * @param {!Window} win
    * @param {!Element} anchorElement
    * @param {!function(!Element, !Element)} injector
+   * @param {!../../../src/layout-rect.LayoutMarginsChangeDef|undefined} margins
```
Comment (lannka) [inline extensions/amp-auto-ads/0.1/placement.js]: `{../../../src/layout-rect.LayoutMarginsChangeDef=}`

File: extensions/amp-auto-ads/0.1/placement.js [hunk 3]
```diff
@@ -82,8 +82,9 @@ export class Placement {
...
    */
-  constructor(win, anchorElement, injector) {
+  constructor(win, anchorElement, injector, margins) {
```
Comment (lannka) [inline extensions/amp-auto-ads/0.1/placement.js]: we follow naming convention for opt param: `opt_margins`

File: extensions/amp-auto-ads/0.1/placement.js (hunk 1 above)
Comment (tlong2) [inline extensions/amp-auto-ads/0.1/placement.js]: Done

File: extensions/amp-auto-ads/0.1/placement.js (hunk 2 above)
Comment (tlong2) [inline extensions/amp-auto-ads/0.1/placement.js]: Done

File: extensions/amp-auto-ads/0.1/placement.js (hunk 3 above)
Comment (tlong2) [inline extensions/amp-auto-ads/0.1/placement.js]: Done

**Comment by lannka** [comment]
@tlong2 seems Github lost Travis status pingback. It doesn't allow me to merge. The easiest way I know is to push an empty commit..

**Comment by tlong2** [comment]
@lannka Have pushed an empty commit

**Pull request merged**